validation-test/stdlib/Collection/LazyFilterCollection.swift

```diff
@@ -0,0 +1,146 @@
+// -*- swift -*-
+
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: %S/../../../utils/gyb %s -o %t/main.swift
+// RUN: %S/../../../utils/line-directive %t/main.swift -- %target-build-swift %t/main.swift -o %t/LazyFilterCollection.swift.a.out
+// RUN: %S/../../../utils/line-directive %t/main.swift -- %target-run %t/LazyFilterCollection.swift.a.out
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import StdlibCollectionUnittest
+
+// Also import modules which are used by StdlibUnittest internally. This
+// workaround is needed to link all required libraries in case we compile
+// StdlibUnittest with -sil-serialize-all.
+import SwiftPrivate
+#if _runtime(_ObjC)
+import ObjectiveC
+#endif
+
+var CollectionTests = TestSuite("Collection")
+
+// Test collection as sequence using value types as elements.
+CollectionTests.addBidirectionalCollectionTests(
+  makeCollection: { (elements: [OpaqueValue<Int>]) in
+    MinimalSequence(elements: elements).lazy.filter { _ in return true }
+  },
+  wrapValue: identity,
+  extractValue: identity,
+  makeCollectionOfEquatable: { (elements: [MinimalEquatableValue]) in
+    MinimalSequence(elements: elements).lazy.filter { _ in return true }
```

Comment:
If you can still type check the "obviously wrong" code, it would be really good if you could understand why it compiles.  You might have found an issue either in the compiler, or in the testing library.